torchci/lib/bot/retryBot.ts

```diff
@@ -1,7 +1,148 @@
 import { Probot } from "probot";
+import { getFlakyJobsFromPreviousWorkflow } from "../jobUtils";
 import { CachedConfigTracker } from "./utils";
 
+const SUCCESS_CONCLUSIONS = ["success"];
 const FAILURE_CONCLUSIONS = ["failure", "cancelled", "timed_out"];
+const ALLOWED_WORKFLOW_PREFIXES: { [key: string]: string[] } = {
+  pytorch: ["lint", "pull", "trunk", "linux-binary", "windows-binary"],
+  vision: [
+    "lint",
+    "Build Linux",
+    "Build Macos",
+    "Build M1",
+    "Tests on Linux",
+    "Tests on macOS",
+  ],
+};
+
+async function retryPreviousWorkflow(
+  ctx: any,
+  owner: string,
+  repo: string,
+  branch: string,
+  workflowName: string,
+  workflowId: number
+) {
+  const flakyJobs = await getFlakyJobsFromPreviousWorkflow(
```

Comment:
What would the rockset query return if the previous workflow still has pending jobs?  

Two potential scenarios:
1.  a job is running very slowly
2. it was manually retried?

Reply:
In both case, it will return nothing because of the status is not failure in the first case and because of the run attempt is not equal to 1 in the second.  The first case should be very rare though as a failed job completes much faster than a successful one, which leaves only the case of the previous job hanging flakily.  IMO, it should ok to deal with it manually when that happens (famous last words lol)